test/kwtool_web/controllers/upload_controller_test.exs

```diff
@@ -0,0 +1,17 @@
+defmodule KwtoolWeb.UploadControllerTest do
+  use KwtoolWeb.ConnCase, async: true
+  alias Kwtool.Accounts.Guardian
+
+  describe "get index/2" do
+    test "returns 200 status", %{conn: conn} do
+      created_user = insert(:user)
+
+      conn =
+        conn
+        |> Guardian.Plug.sign_in(created_user)
```

Comment:
To avoid having to include `Kwtool.Accounts.Guardian` each time, a helper could be created in `test/support` (e.g. `KwtoolWeb.Support.ConnHelper` which is then aliased in `KwtoolWeb.ConnCase`):

```elixir
defmodule KwtoolWeb.Support.ConnHelper
  alias Kwtool.Accounts.Guardian

  def with_signed_in_user(%Conn{} = conn, user) do
    Guardian.Plug.sign_in(conn, user)
  end
end
```

So that it can be used as:

```elixir
current_user = insert(:user)

conn =
  conn
  |> with_signed_in_user(current_user)
```

Reply:
Yes, I updated the code using conn helper.